RecoHI/HiJetAlgos/plugins/HiPuRhoProducer.cc

```diff
@@ -0,0 +1,615 @@
+      vrms0[vi] = initialValue;
+      vrho0[vi] = initialValue;
+    }
+  }
+
+  for (int i = ietamin_; i < ietamax_ + 1; i++) {
+    emean_[i] = 0.;
+    emean2[i] = 0.;
+    esigma_[i] = 0.;
+    ntowers[i] = 0;
+
+    eTop4_[i] = {0., 0., 0., 0.};
+  }
+
+  for (auto const& input_object : coll) {
+    const reco::Candidate* originalTower = inputs_[input_object.user_index()];
+    double original_Et = originalTower->et();
+    const CaloTower* ctc = dynamic_cast<const CaloTower*>(originalTower);
+    int ieta0 = ctc->id().ieta();
+
+    if (original_Et > eTop4_[ieta0][0]) {
+      eTop4_[ieta0][3] = eTop4_[ieta0][2];
+      eTop4_[ieta0][2] = eTop4_[ieta0][1];
+      eTop4_[ieta0][1] = eTop4_[ieta0][0];
+      eTop4_[ieta0][0] = original_Et;
+    } else if (original_Et > eTop4_[ieta0][1]) {
+      eTop4_[ieta0][3] = eTop4_[ieta0][2];
+      eTop4_[ieta0][2] = eTop4_[ieta0][1];
+      eTop4_[ieta0][1] = original_Et;
+    } else if (original_Et > eTop4_[ieta0][2]) {
+      eTop4_[ieta0][3] = eTop4_[ieta0][2];
+      eTop4_[ieta0][2] = original_Et;
+    } else if (original_Et > eTop4_[ieta0][3]) {
+      eTop4_[ieta0][3] = original_Et;
+    }
+
+    emean_[ieta0] = emean_[ieta0] + original_Et;
+    emean2[ieta0] = emean2[ieta0] + original_Et * original_Et;
+    if (ieta0 - ietaold != 0) {
+      ntowers[ieta0] = 1;
+      ietaold = ieta0;
+    } else {
+      ntowers[ieta0]++;
+    }
+  }
+
+  for (auto const& gt : geomtowers_) {
+    int it = gt.first;
+
+    int vi = it - 1;
+
+    if (it < 0)
+      vi = nEtaTow_ + it;
+
+    double e1 = emean_[it];
+    double e2 = emean2[it];
+    int nt = gt.second - ntowersWithJets_[it];
+
+    if (vi < nEtaTow_) {
+      vngeom[vi] = gt.second;
+      vntow[vi] = nt;
+    }
+
+    LogDebug("PileUpSubtractor") << " ieta: " << it << " number of towers: " << nt << " e1: " << e1 << " e2: " << e2
+                                 << "\n";
+
+    if (nt > 0) {
+      if (postOrphan_) {
+        if (nt > (int)minimumTowersFraction_ * (gt.second)) {
+          emean_[it] = e1 / (double)nt;
+          double eee = e2 / (double)nt - e1 * e1 / (double)(nt * nt);
+          if (eee < 0.)
+            eee = 0.;
+          esigma_[it] = nSigmaPU_ * sqrt(eee);
+
+          uint32_t numToCheck = std::min(int(eTop4_[it].size()), nt - (int)minimumTowersFraction_ * (gt.second));
+
+          for (unsigned int lI = 0; lI < numToCheck; ++lI) {
+            if (eTop4_[it][lI] >= emean_[it] + towSigmaCut_ * esigma_[it] && towSigmaCut_ > 0) {
+              e1 -= eTop4_[it][lI];
+              nt -= 1;
+            } else
+              break;
+          }
+
+          if (e1 < .000000001)
+            e1 = 0;
+        }
+      }
+
+      emean_[it] = e1 / (double)nt;
+      double eee = e2 / nt - e1 * e1 / (nt * nt);
+      if (eee < 0.)
+        eee = 0.;
+      esigma_[it] = nSigmaPU_ * sqrt(eee);
+
+      double etaWidth = etaedge[abs(it)] - etaedge[abs(it) - 1];
+      if (etaWidth < 0)
+        etaWidth *= -1.;
```

Comment:
isn't this the same as 
```suggestion
      double etaWidth = std::abs(etaedge[abs(it)] - etaedge[abs(it) - 1]);
```
?